Tests/CreateAPITests/Expected/petstore-no-package/Entities.swift

```diff
@@ -39,6 +39,37 @@ public struct Error: Codable {
     }
 }
 
+public struct Dog: Codable {
+    /// A pet title
+    ///
+    /// A pet description
+    public var pet: Pet
+    public var breed: Breed?
+
+    public enum Breed: String, Codable, CaseIterable {
+        case large = "Large"
+        case medium = "Medium"
+        case small = "Small"
+    }
+
+    public init(pet: Pet, breed: Breed? = nil) {
+        self.pet = pet
+        self.breed = breed
+    }
+
+    public init(from decoder: Decoder) throws {
+        let values = try decoder.container(keyedBy: StringCodingKey.self)
+        self.pet = try Pet(from: decoder)
+        self.breed = try values.decodeIfPresent(Breed.self, forKey: "breed")
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var values = encoder.container(keyedBy: StringCodingKey.self)
+        try values.encode(pet, forKey: "pet")
+        try values.encodeIfPresent(breed, forKey: "breed")
+    }
```

Comment:
Great work on `Identifiable`. That's a nice addition for all SwiftUI users.

Yeah, `Dog`'s encoder is incorrect. I forgot to make it match the initializer.

```swift
    public init(from decoder: Decoder) throws {
         let values = try decoder.container(keyedBy: StringCodingKey.self)
         // This code reads properties from the existing decoder (not nested) 
         self.pet = try Pet(from: decoder)
         self.breed = try values.decodeIfPresent(Breed.self, forKey: "breed")
     }

     public func encode(to encoder: Encoder) throws {
         var values = encoder.container(keyedBy: StringCodingKey.self)
         // This should do the same: `pet.encode(to: encoder)`
         try values.encode(pet, forKey: "pet")
         try values.encodeIfPresent(breed, forKey: "breed")
     }
```

It would be nice to cover this piece with a bit uint test that checks that encoding/decoding pair actually works regardless of the implementation details.